miio/integrations/petfooddispenser/status.py

```diff
@@ -0,0 +1,76 @@
+import enum
+from typing import Any, Dict
+
+from miio.miot_device import DeviceStatus
+
+class FoodStatus(enum.Enum):
+    Normal = 0
+    Low = 1
+    Empty = 2
+
+class PowerState(enum.Enum):
+    Mains = 0
+    Battery = 1
+
+class PetFoodDispenserStatus(DeviceStatus):
+    """Container for status reports from the Pet Feeder / Smart Pet Food Dispenser."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.data = data
+
+    @property
+    def food_status(self) -> str:
+        """Current food status / level."""
+        return FoodStatus(self.data["food_status"])
+
+    @property
+    def feed_plan(self) -> bool:
+        """Automatic feeding status."""
+        return bool(self.data["feed_plan"])
+
+    @property
+    def door_status(self) -> bool:
+        """Food bin door status."""
+        return bool(self.data["door_status"])
+
+    @property
+    def feed_today(self) -> int:
+        """Units of food dispensed today."""
+        raise NotImplementedError("Device does not currently return this value correctly.")
```

Comment:
I think it'd be better to either return the incorrect value (and log an error) or remove not implemented properties altogether.

Reply:
It's a tough one. Returning bad data seems worse than not returning the property at all.
I didn't want the implementation to look incomplete but nor did I want to return bad data.

It is your project though so given the 2 choices I'll go with remove :+1:

Reply:
As the data is still available in the response payloads, anyone can go and check it out in case they start returning more proper values, right? Not having properties raising unexpected exceptions makes it possible to use the whole API without having to dig into the source, so that's why I prefer it that way.